### PR TITLE
[RM Ch1] Draft Networking language for Baldy

### PR DESCRIPTION
**Pull request by markshostak**
This material applies to Issue 1037, and is intended to kick off content generation for Baldy.

### CODE REVIEW

**Comment by markshostak** [comment]
> 
> @markshostak the layering diagram seems to be missing....

@walterkozlowski Hi Walter, It's just hiding! :-) It will show up once everything is merged. I couldn't talk github into uploading it as part of the same PR, so it will magically appear once everything is merged. Thanks, -Mark

**Comment by markshostak** [comment]
@TFredberg Hi Tomas, I incorporated your feedback/edits. Thanks, -Mark

**Comment by rabi-abdel** [comment]
looks good. one more approval to merge.

**Comment by pgoyal01** [comment]
The content is what @TFredberg wrote in a conversation therefore it needs additional approvers.

**Comment by markshostak** [comment]
> 
> The content is what @TFredberg wrote in a conversation therefore it needs additional approvers.

That's inaccurate.

File: doc/ref_model/chapters/chapter01.md [hunk 1]
```diff
@@ -238,7 +243,98 @@ The ratings are as follows:
...
+
+The following subsections, Executive Summary, Strategy Objectives, Networking Requirements and Initial Approach, respectively, are intended to provide an overview of CNTT's vision for networking and intent, objectives, requirements and their supporting rationale, as well as initial development approach.
+
+The networking within an NFVI, fabric or otherwise, is an area where there is significant variability across implementations. Leaf-Spine topology is well established, however, after topology there are countless decisions an Operator needs to make. Differences arise from many aspects, for example, is the solution layer-2 or layer-3; is the routing static or dynamic; what mechanism is used for encapsulation; what mechanism is used for isolation; does it support SR-IOV; does it support DPDK; does it employee SmartNICs; does it employ distributed control or a centralized control driving a programmable fabric; and the list continues. The multitude of permutations enable Operators to design (or procure) a fabric/networking solution that's optimized for their needs, whether their needs are minimal, very extensive or somewhere in between.
```
Comment (kedmison) [inline doc/ref_model/chapters/chapter01.md]: employee -> employs

File: doc/ref_model/chapters/chapter01.md [hunk 2]
```diff
@@ -238,7 +243,98 @@ The ratings are as follows:
...
+
+For CNTT, a strategy is needed that affords Operators the performance, flexibility, availability, maintainability and scalability their business requires, yet doesn't require OPNFV to design, manage and test prohibitive numbers of networking solutions. // relates to Petar's #2
```
Comment (kedmison) [inline doc/ref_model/chapters/chapter01.md]: Request removal of the comment referring to a CNTT contributor

File: doc/ref_model/chapters/chapter01.md (hunk 1 above)
Comment (kedmison) [reply doc/ref_model/chapters/chapter01.md]: enable Operators to design -> enable NFVI architects to design... would like to ensure NFVI vendors are included.

File: doc/ref_model/chapters/chapter01.md [hunk 3]
```diff
@@ -238,7 +243,98 @@ The ratings are as follows:
...
+<a name="1.9.2"></a>
+## 1.9.2 Networking Strategy Objectives
+
+This section catalogs CNTT's high-level objectives for the Networking and Fabric Strategy. 
+
+> _List needs to be prioritized; expect additional objectives to be added, as they arise. This represents the _What_, not the _How_.
+
+1. The implementation of Networking inside the HW Layer should not be visible to the VNF/CNF and should preferably not even be visible to the IaaS/CaaS // TF #2
+1. Provide networks for L3 tenant, GWs, SDS, etc.
+1. CLEANLY decouple interface/reference points between CNTT constituencies
```
Comment (kedmison) [inline doc/ref_model/chapters/chapter01.md]: Please remove all-caps use for "Cleanly".

File: doc/ref_model/chapters/chapter01.md (hunk 1 above)
Comment (markshostak) [reply doc/ref_model/chapters/chapter01.md]: Done.
Note: employee -> employ (no s)
Note: incorporated intent; language slightly modified

File: doc/ref_model/chapters/chapter01.md (hunk 2 above)
Comment (markshostak) [reply doc/ref_model/chapters/chapter01.md]: Done
Note: Multiple reference deleted

File: doc/ref_model/chapters/chapter01.md (hunk 3 above)
Comment (markshostak) [reply doc/ref_model/chapters/chapter01.md]: Done

**Comment by walterkozlowski** [comment]
Hi All.   
Generally, this is a good plan and a good starting point, considering the timeframe we need to stabilise the contents for Baldy.  Obviously, this is only starting point, and I do hope that the content we have started developing with the team will be part of the work identified here.  

However, I would like to propose some minor changes.  
@rabi-abdel What is the best way to introduce changes, now you merged this contents into master?  I do not want to create too much fuss about.

Some changes:
1. I am not sure whether we need Sec 1.9.3 Requirements at this stage. At this level of ambihguity , they will generate too much confusion, and will not add any value. I strongly propose to remove the contents from 1.9.3 and replace it by "To be worked out in LA meetings and the follow-up work". 
2. If we decide after all to go ahead with the contents for 1.9.3 for Baldy, we need to completely rewrite some of these requirements as they are not stated in the industry accepted terminology, and are missing a lot of important aspects of networking. Hence we would need to add much more contents, for which we do not have much time, I am afraid.

**Comment by markshostak** [comment]
@walterkozlowski Hi Walter, WRT how to make changes/enhancements, it's the same way we make all changes and enhancements, just open a PR. This was just the first PR, with the expectation there would be more to follow.

WRT 1.9.3, I'm thinking we do need high-level requirements at this time, to guide the solution we create. However, that doesn't mean they have to be exhaustive for Baldy. It's perfectly acceptable to incorporate them as we recognize them.

**Comment by walterkozlowski** [comment]
@markshostak Hi Mark, if we want to have requirements, we need to do A LOT of changes and amendments, and make sure we are clear and we use industry terminology (so we can also discuss them with other communities like ONF).  I will raise a PR to get us work on Requirements, and will include changes as I propose.  Unless of course other people are inclined more  towards not doing Requirements in the Baldy timeframe.

**Comment by rabiabdel** [comment]
@walterkozlowski @markshostak with large PRs such as this, it is often easier to merge an initial content and have subsequent PRs to modify it. So I would suggest creating a new PR to address the remaining comments if any. Thanks

**Comment by ASawwaf** [comment]
Thanks, @markshostak for the writeup as usual impressive :) , shall I consider what is written down in this PR is Outlines for Networking / SDN/Fabric  as I don't see any contradicting with what @walterkozlowski mentioned in issue  #1037 

we need to develop contents under these outlines

**Comment by ASawwaf** [comment]
@walterkozlowski @markshostak , for section 1.9.3 Network req. we need to add more and more here

**Comment by oyayan** [comment]
@markshostak good starting point and covers key points of the comments. I believe requirements section is relevant here but also agree that it needs to be developed.

**Pull request merged**